samples/nextjs/scripts/templates/component-src.ts

```diff
@@ -16,7 +16,7 @@ type ${componentName}Props = StyleguideComponentProps & {
 const ${componentName} = (props: ${componentName}Props): JSX.Element => (
   <div>
     <p>${componentName} Component</p>
-    <Text field={props.fields.heading} />
+    <Text field={props.fields?.heading} />
```

Comment:
We should use the `withDatasourceCheck` HOC here, which provides datasource resiliency for Horizon. See `ContentBlock.tsx` for example.

Reply:
Yes makes sense, done.